fpmsyncd/routesync.cpp

```diff
@@ -1222,3 +1269,183 @@ string RouteSync::getNextHopWt(struct rtnl_route *route_obj)
 
     return result;
 }
+
+bool RouteSync::sendOffloadReply(struct nlmsghdr* hdr)
+{
+    SWSS_LOG_ENTER();
+
+    if (hdr->nlmsg_type != RTM_NEWROUTE)
+    {
+        return false;
+    }
+
+    struct rtmsg *rtm = static_cast<struct rtmsg*>(NLMSG_DATA(hdr));
+
+    rtm->rtm_flags |= RTM_F_OFFLOAD;
+
+    if (!m_fpmInterface)
+    {
+        SWSS_LOG_ERROR("Cannot send offload reply to zebra: FPM is disconnected");
+        return false;
+    }
+
+    // Send to zebra
+    if (!m_fpmInterface->send(hdr))
+    {
+        SWSS_LOG_ERROR("Failed to send reply to zebra");
+        return false;
+    }
+
+    return true;
+}
+
+bool RouteSync::sendOffloadReply(struct rtnl_route* route_obj)
+{
+    SWSS_LOG_ENTER();
+
+    nl_msg* msg{};
+    rtnl_route_build_add_request(route_obj, NLM_F_CREATE, &msg);
+
+    auto nlMsg = makeUniqueWithDestructor(msg, nlmsg_free);
+
+    return sendOffloadReply(nlmsg_hdr(nlMsg.get()));
+}
+
+void RouteSync::setSuppressionEnabled(bool enabled)
+{
+    SWSS_LOG_ENTER();
+
+    m_isSuppressionEnabled = enabled;
+
+    SWSS_LOG_NOTICE("Pending routes suppression is %s", (m_isSuppressionEnabled ? "enabled": "disabled"));
+}
+
+void RouteSync::onRouteResponse(const std::string& key, const std::vector<FieldValueTuple>& fieldValues)
+{
+    IpPrefix prefix;
+    std::string vrfName;
+    std::string protocol;
+
+    bool isSetOperation{false};
+    bool isSuccessReply{false};
+
+    if (!isSuppressionEnabled())
+    {
+        return;
+    }
+
+    auto colon = key.find(':');
+    if (colon != std::string::npos && key.substr(0, colon).find(VRF_PREFIX) != std::string::npos)
+    {
+        vrfName = key.substr(0, colon);
+        prefix = IpPrefix{key.substr(colon + 1)};
+    }
+    else
+    {
+        prefix = IpPrefix{key};
+    }
+
+    for (const auto& fieldValue: fieldValues)
+    {
+        std::string field = fvField(fieldValue);
+        std::string value = fvValue(fieldValue);
+
+        if (field == "err_str")
+        {
+            isSuccessReply = (value == "SWSS_RC_SUCCESS");
+        }
+        else if (field == "protocol")
+        {
+            // If field "protocol" is present in the field values then
+            // it is a SET operation. This field is absent only if we are
+            // processing DEL operation.
+            isSetOperation = true;
```

Comment:
Dont we get Del operation in the event? why should we assume this?

Reply:
If we get an response for a delete operation we don't get original FVs. We use this knowledge to filter out DEL events we are not interested in.

ResponsePublisher for reference: https://github.com/sonic-net/sonic-swss/blob/master/orchagent/response_publisher.cpp#L106